pkgs/development/tools/godot/mono/default.nix

```diff
@@ -0,0 +1,94 @@
+{ godot
+, lib
+, callPackage
+, mkNugetDeps
+, mkNugetSource
+, mono
+, dotnet-sdk
+, writeText
+}:
+
+godot.overrideAttrs (self: base: {
+  pname = "godot-mono";
+
+  passthru = let bp = base.passthru; in bp // {
+    sconsFlagSet = bp.sconsFlagSet // {
+      module_mono_enabled = true;
+      mono_prefix = mono;
+    };
+
+    buildDescription = "mono build";
+
+    glue = callPackage ./glue.nix {};
+
+    make-deps = callPackage ./make-deps.nix {};
+  };
+
+  nativeBuildInputs = base.nativeBuildInputs ++ [ mono dotnet-sdk ];
+
+  nugetDeps = mkNugetDeps { name = "deps"; nugetDeps = import ./deps.nix; };
+
+  nugetSource =
+    mkNugetSource {
+      name = "${self.pname}-nuget-source";
+      description = "A Nuget source with dependencies for ${self.pname}";
+      deps = [ self.nugetDeps ];
+    };
+
+  nugetConfig = writeText "NuGet.Config" ''
+    <?xml version="1.0" encoding="utf-8"?>
+    <configuration>
+      <packageSources>
+        <add key="${self.pname}-deps" value="${self.nugetSource}/lib" />
+      </packageSources>
+    </configuration>
+  '';
+
+  configureSteps = [ "configureHomeStep" "configureNugetStep" "stageGlueStep" ];
```

Comment:
This could easily be controlled via a boolean input to run the phase or not. Then we conditionally build a string and run it as configurePhase. That's way less complex and easier for basically creating a directory and exporting a variable for each step.

Reply:
It seems more flexible to just let the overriding derivation specify the steps they want to run.  They may want to add steps or rearrange steps, not just filter out one of them.  Simply having one big configure phase with a flag that disables sections of it presumes a lot about what an overrider might want to do.

Reply:
I _can_ utilize the `postConfigureHooks` rather than overriding `configurePhase`, and now that I think about it that would be better since overriding `configurePhase` replaces whatever the godot derivation might do in the future.  However, I don't want to just do this:

```nix
{
  postConfigureHooks = [
    (''
       # contents of configureHomeStep
     '')
    (''
       # contents of configureNugetStep
     '')
    (''
       # ccontents of stageGlueStep
     '')
  ];
}
```
Because the overriding derivation (e.g. make-deps.nix) can't easily say "do all the steps except for the one that configures nuget".

How about this?
```nix
{
  # ...
  passthru = {
    # ...
    configureSteps = [ "configureHomeStep" "configureNugetStep" "stageGlueStep" ];
    # ...
  };

  postConfigureHooks = builtins.map (s: "eval \"\$${s}\"") self.passthru.configureSteps;

  configureHomeStep = ...;
  configureNugetStep = ...;
  stageGlueStep = ...;
  # ...
}
```
Moved configureSteps into passthru since it's no longer needed directly within the configurePhase, and then removed the override of configurePhase and replaced it with a list of postConfigureHooks (which just evals each of the steps)